### PR TITLE
fix #11: ws socket error after upgrading to 4.0

### LINKED ISSUE #11: Upgrade ws package
The latest ws package is now 4.0, but this project is still using ws 2.0.
I did some basic benchmark with sending 20000 rpc to add two numbers in parallel. The result of using ws 2.0 is 1300ms (wamp.rt uses ws 0.8 with 2000ms), while using ws 4.0 (without check protocol) is 750ms. This is a huge impact and it is worth for a dependency upgrade.

But I notice that ws api has been changed a bit. There an error of obtain ip address during startup. Here is the description of the api change https://github.com/websockets/ws/pull/1099

### PR DESCRIPTION
**Pull request by stanleyxu2005**
also allow to specify host (by default: 0.0.0.0)
also change node module version from fixed version number to at least (will not upgrade to next major release)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -33,22 +33,22 @@
...
+    "mocha": "^2.5.3",
+    "run-sequence": "^1.2.2",
+    "tslint": "^3.13.0",
+    "typescript": "^1.8.10"
   },
   "dependencies": {
-    "debug": "2.2.0",
-    "ws": "4.0.0"
+    "debug": "^2.2.0",
+    "ws": "^4.0.0"
```
Comment (ivaylopivanov) [inline package.json]: I would prefer to not have it this way.. We already had similar issue with ws https://github.com/ivaylopivanov/wamp-server/pull/5

File: src/server.ts [hunk 2]
```diff
@@ -34,8 +42,9 @@ class Server {
    */
   constructor(private options: OptionsInterface) {
     const REALMS: string | string[] = options.realms;
+    this.host = options.host
```
Comment (ivaylopivanov) [inline src/server.ts]: lets have localhost as default value

File: package.json (hunk 1 above)
Comment (ivaylopivanov) [reply package.json]: I have enabled Greenkeeper to take care of the dependencies.

**Comment by stanleyxu2005** [comment]
Let me create another one

**Pull request closed without merging**